src/builder.ts

```diff
@@ -490,11 +490,13 @@ export class SchemaBuilder {
         }
       });
     } else {
-      Object.keys(members).forEach((key) => {
-        values[key] = {
-          value: members[key],
-        };
-      });
+      Object.keys(members)
+        .filter((key) => isNaN(+key)) // Ensures we only consider enum keys in case a TypeScript enum was passed as the members property.
```

Comment:
Symbols can technically be keys of a mapping, though probably rare we may want to guard against that case.

Reply:
We may also want to check not just if it's not a string, but also that `members[key] === members[members[key]]` which would indicate that it's actually a typescript enum, rather than a user-defined key in a normal object that happens to be an invalid value. 

Along those lines, I think it'd be worthwhile to add GraphQL's `assertValidName` guard on the `key` of the mapping similar to how we do it elsewhere for the object & field names.

Reply:
> Symbols can technically be keys of a mapping, though probably rare we may want to guard against that case.

Do you mean it's possible that Symbols are the keys of `Object.keys` and that we should guard against having them as member keys as input for `GraphQLEnumType`?

Reply:
Actually, I thought Symbols could be values of an enum but they can't, so you can ignore that.

Reply:
I added an `assertValidName` check to ensure the enum name is valid in the end. The suggest members check is not really feasible because that check would only work for enums that have a numeric value, and since enum names of a TS enum are always strings the `isNaN` check is actually the more reliable and simple way I found to ensure we grab the right keys.

I also improve the test assertions to ensure both names and values are what is to be expected.